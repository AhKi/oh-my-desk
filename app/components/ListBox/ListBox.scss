.ListBox {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  align-items: stretch;

  border-radius: .25rem;
  border: 1px solid rgba(0,0,0,.125);
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,sans-serif;
  overflow: scroll;
  height: 200px;
  min-width: 350px;
}
