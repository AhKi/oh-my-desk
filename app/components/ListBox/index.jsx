import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import ListItem from 'components/ListItem';
import './ListBox.scss';

const propTypes = {
	list: PropTypes.array, // eslint-disable-line
	selectedId: PropTypes.string,
	onSelectItem: PropTypes.func,
};

const defaultProps = {
	list: [],
	selectedId: '',
	onSelectItem() {},
};

class ListBox extends React.Component {
	render() {
		const { list, selectedId, onSelectItem } = this.props;
		const listBoxClassName = cx('ListBox');

		return (
			<ul className={listBoxClassName}>
				{list.map(item => (<ListItem
					key={item.id}
					item={item}
					selectedId={selectedId}
					onSelectItem={onSelectItem}
				/>))}
			</ul>
		);
	}
}

ListBox.propTypes = propTypes;
ListBox.defaultProps = defaultProps;

export default ListBox;
