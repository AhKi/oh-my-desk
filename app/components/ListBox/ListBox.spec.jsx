import React from 'react';
import { shallow } from 'enzyme';

import ListBox from './';

describe('<ListBox />', () => {
  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<ListBox list={[{ id: 1 }]} />);

    expect(wrapper).toMatchSnapshot();
  });
});
