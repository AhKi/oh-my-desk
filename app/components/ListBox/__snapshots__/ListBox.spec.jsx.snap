// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListBox /> should match to snapshot when render default 1`] = `
<ul
  className="ListBox"
>
  <ListItem
    item={
      Object {
        "id": 1,
      }
    }
    key="1"
    onSelectItem={[Function]}
    selectedId=""
  />
</ul>
`;
